### DISCUSSION TITLE
How to delete file in target directory and update in chezmoi repo?

### DISCUSSION THREAD

**Discussion by allworldg** [twpayne/chezmoi#2810] [Q&A] [answered]
I try to use chezmoi to manage my neovim config directory . I often do something like a following example:
1. update my neovim config ( edit files , add files and delete files in ~/.config/nvim ) .
2. Sometimes I want to make a backup . so `chezmoi add ~/.config/nvim`.

The problem is when i delete files and use add command , chezmoi can not delete same files in repo . I know this strategy is reasonable . So are there any suggestions to solve it ? I don't want to delete whole dir everytime .

**Comment by twpayne** [comment, marked as answer]
The easiest way is probably to tell chezmoi to forget the directory and then re-add it:

```console
$ chezmoi forget ~/.config/nvim
$ chezmoi add ~/.config/nvim
```

**Reply by allworldg** [reply]
Thank you . Maybe I should manage nvim config without chezmoi , just use git .

**Reply by twpayne** [reply]
You could, but chezmoi is just a wrapper around git already.

**Reply by allworldg** [reply]
Yes , my dotfile still use it .

**Reply by Lockszmith-GH** [reply]
A git repo as an external resource in chezmoi can probably give you a good mix of both approaches.

**Reply by allworldg** [reply]
@Lockszmith-GH  Sorry, I don't know the meaning of " external resource in chezmoi " . Maybe create a submodule ? Could you explain it ? Thank you .

**Reply by bradenhilton** [reply]
@allworldg https://www.chezmoi.io/reference/special-files-and-directories/chezmoiexternal-format/

**Reply by Lockszmith-GH** [reply]
what @bradenhilton said, specifically `type = git-repo`

**Reply by allworldg** [reply]
Thank you . I will try .
